### PR TITLE
unix-privesc-check: init at 1.4

### PR DESCRIPTION
**Pull request by D3vil0p3r**
## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/un/unix-privesc-check/package.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  pname = "unix-privesc-check";
+  version = "1.4";
+
+  src = fetchurl {
+    url = "http://pentestmonkey.net/tools/unix-privesc-check/${finalAttrs.pname}-${finalAttrs.version}.tar.gz";
+    hash = "sha256-4fhef2n6ut0jdWo9dqDj2GSyHih2O2DOLmGBKQ0cGWk=";
+  };
+
+  installPhase = ''
+    install -Dm 755 unix-privesc-check $out/bin/unix-privesc-check
```
Comment (Mindavi) [inline pkgs/by-name/un/unix-privesc-check/package.nix:16]: Also runHook preInstall and runHook postInstall

**Review by Mindavi — APPROVED** [review]
LGTM other than the one comment

File: pkgs/by-name/un/unix-privesc-check/package.nix (hunk 1 above)
Comment (D3vil0p3r) [reply pkgs/by-name/un/unix-privesc-check/package.nix:17]: Fixed

**Comment by D3vil0p3r** [comment]
> LGTM other than the one comment

Fixed

File: pkgs/by-name/un/unix-privesc-check/package.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+, stdenvNoCC
+, fetchurl
+}:
+
+stdenvNoCC.mkDerivation (finalAttrs: {
+  pname = "unix-privesc-check";
+  version = "1.4";
+
+  src = fetchurl {
+    url = "http://pentestmonkey.net/tools/unix-privesc-check/${finalAttrs.pname}-${finalAttrs.version}.tar.gz";
```
Comment (SuperSandro2000) [inline pkgs/by-name/un/unix-privesc-check/package.nix:11]:
```suggestion
    url = "https://pentestmonkey.net/tools/unix-privesc-check/unix-privesc-check-${finalAttrs.version}.tar.gz";
```
this should not use finalAttrs.pname as a change in pname would bery likely break the URL

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/by-name/un/unix-privesc-check/package.nix (hunk 2 above)
Comment (D3vil0p3r) [reply pkgs/by-name/un/unix-privesc-check/package.nix]: Fixed

**Comment by MinerSebas** [comment]
This is missing depenencies:
```
$ nix run github:D3vil0p3r/nixpkgs/patch-8#unix-privesc-check -- standard
Assuming the OS is: linux
which: no file in (/run/wrappers/bin:/home/sebastian/.nix-profile/bin:/nix/profile/bin:/home/sebastian/.local/state/nix/profile/bin:/etc/profiles/per-user/sebastian/bin:/nix/var/nix/profiles/default/bin:/run/current-system/sw/bin:/usr/bin:/bin:/sbin:/usr/sbin)
ERROR: Dependend program 'file' is mising.  Can't run.  Sorry!
```
Once i enter a nix-shell taht contains toybox, does the script succesfully run.

**Comment by D3vil0p3r** [comment]
> This is missing depenencies:
> 
> ```
> $ nix run github:D3vil0p3r/nixpkgs/patch-8#unix-privesc-check -- standard
> Assuming the OS is: linux
> which: no file in (/run/wrappers/bin:/home/sebastian/.nix-profile/bin:/nix/profile/bin:/home/sebastian/.local/state/nix/profile/bin:/etc/profiles/per-user/sebastian/bin:/nix/var/nix/profiles/default/bin:/run/current-system/sw/bin:/usr/bin:/bin:/sbin:/usr/sbin)
> ERROR: Dependend program 'file' is mising.  Can't run.  Sorry!
> ```
> 
> Once i enter a nix-shell that contains toybox, the script succesfully runs.

If I add `toybox` as dependency in `buildInputs`, and I build the package, I get:
```
this derivation will be built:
  /nix/store/446zp6r2imix1qihkk8rdclzahnyx58w-unix-privesc-check-1.4.drv
building '/nix/store/446zp6r2imix1qihkk8rdclzahnyx58w-unix-privesc-check-1.4.drv'...
Running phase: unpackPhase
unpacking source archive /nix/store/1nssihpc7670iz9c5ngw50446w4syvra-unix-privesc-check-1.4.tar.gz
tar: Unknown option 'warning=no-timestamp' (see "tar --help")
do not know how to unpack source archive /nix/store/1nssihpc7670iz9c5ngw50446w4syvra-unix-privesc-check-1.4.tar.gz
error: builder for '/nix/store/446zp6r2imix1qihkk8rdclzahnyx58w-unix-privesc-check-1.4.drv' failed with exit code 1;
       last 4 log lines:
       > Running phase: unpackPhase
       > unpacking source archive /nix/store/1nssihpc7670iz9c5ngw50446w4syvra-unix-privesc-check-1.4.tar.gz
       > tar: Unknown option 'warning=no-timestamp' (see "tar --help")
       > do not know how to unpack source archive /nix/store/1nssihpc7670iz9c5ngw50446w4syvra-unix-privesc-check-1.4.tar.gz
       For full logs, run 'nix log /nix/store/446zp6r2imix1qihkk8rdclzahnyx58w-unix-privesc-check-1.4.drv'.
```

**Comment by h7x4** [comment]
Maybe you could try wrapping it into the programs `$PATH` instead?

**Comment by K900** [comment]
You want to use `which`, not `toybox`.

**Comment by K900** [comment]
The entire thing is just a shell script, which relies on ambient binaries from the environment. It should be wrapped with the right inputs, or ran through resholve, so it is self-contained and has correct dependencies tracked.

**Review by K900 — DISMISSED** [review]
Also does not eval.

**Comment by D3vil0p3r** [comment]
Thank you very much guys. Thank you @K900 for instructing me about resholve. @MinerSebas now it should work in your case.

**Comment by MinerSebas** [comment]
> @MinerSebas now it should work in your case.

Still doesn't work :shrug: 
```
nix run github:D3vil0p3r/nixpkgs/patch-8#unix-privesc-check -- standard
Assuming the OS is: linux
/nix/store/g7pq0y4hxj4bsmpfq3yz5r8n925z6rzc-which-2.21/bin/which: no file in (/run/wrappers/bin:/home/sebastian/.nix-profile/bin:/nix/profile/bin:/home/sebastian/.local/state/nix/profile/bin:/etc/profiles/per-user/sebastian/bin:/nix/var/nix/profiles/default/bin:/run/current-system/sw/bin:/usr/bin:/bin:/sbin:/usr/sbin)
ERROR: Dependend program 'file' is mising.  Can't run.  Sorry!
```

**Comment by D3vil0p3r** [comment]
@MinerSebas it is strange because as you can see, which is inside the code as dependency.

**Comment by MinerSebas** [comment]
> @MinerSebas it is strange because as you can see, which is inside the code as dependency.

The script isn't complaining that it can't find `which`,
`which` complains that it cant find the program `file`.
But the essence is still true, `file` is in the inputs.

```
$ nix shell nixpkgs#file
$ nix run github:D3vil0p3r/nixpkgs/patch-8#unix-privesc-check -- standard
Assuming the OS is: linux
/nix/store/g7pq0y4hxj4bsmpfq3yz5r8n925z6rzc-which-2.21/bin/which: no strings in (/nix/store/kv5fs6gk153xkg4qzrnk0cs8myrc1wki-file-5.45/bin:/nix/store/g4mpqylz5blhnnvlh25jb2f3zmddnc8a-file-5.45-man/bin:/run/wrappers/bin:/home/sebastian/.nix-profile/bin:/nix/profile/bin:/home/sebastian/.local/state/nix/profile/bin:/etc/profiles/per-user/sebastian/bin:/nix/var/nix/profiles/default/bin:/run/current-system/sw/bin:/usr/bin:/bin:/sbin:/usr/sbin)
ERROR: Dependend program 'strings' is mising.  Can't run.  Sorry!
```
Using a nix shell that provides `file`, then complains about a missing `strings`, which also should be handled by the inclusion of `binutils`.

Honestly I am out of my depth here. unless someone else can reproduce it, I have no idea how to debug this.

**Comment by D3vil0p3r** [comment]
> > @MinerSebas it is strange because as you can see, which is inside the code as dependency.
> 
> The script isn't complaining that it can't find `which`, `which` complains that it cant find the program `file`. But the essence is still true, `file` is in the inputs.
> 
> ```
> $ nix shell nixpkgs#file
> $ nix run github:D3vil0p3r/nixpkgs/patch-8#unix-privesc-check -- standard
> Assuming the OS is: linux
> /nix/store/g7pq0y4hxj4bsmpfq3yz5r8n925z6rzc-which-2.21/bin/which: no strings in (/nix/store/kv5fs6gk153xkg4qzrnk0cs8myrc1wki-file-5.45/bin:/nix/store/g4mpqylz5blhnnvlh25jb2f3zmddnc8a-file-5.45-man/bin:/run/wrappers/bin:/home/sebastian/.nix-profile/bin:/nix/profile/bin:/home/sebastian/.local/state/nix/profile/bin:/etc/profiles/per-user/sebastian/bin:/nix/var/nix/profiles/default/bin:/run/current-system/sw/bin:/usr/bin:/bin:/sbin:/usr/sbin)
> ERROR: Dependend program 'strings' is mising.  Can't run.  Sorry!
> ```
> 
> Using a nix shell that provides `file`, then complains about a missing `strings`, which also should be handled by the inclusion of `binutils`.
> 
> Honestly I am out of my depth here. unless someone else can reproduce it, I have no idea how to debug this.

What if you store the Nix code in a package.nix file and run:
```
nix-build -E 'with import <nixpkgs> {}; callPackage ./package.nix {}'
```

**Comment by MinerSebas** [comment]
Succesfully builds, but still complain at runtime:

```
$ nix-build -E 'with import <nixpkgs> {}; callPackage ./package.nix {}'

these 8 derivations will be built:
  /nix/store/0ci8kn552dgki9bvvw4g6j72i1fbxpaq-unix-privesc-check-unresholved-1.4.drv
  /nix/store/3s8q8yr21pz1wqpw0iczfv1cdyvnjzpq-postgresql-15.5-binlore.drv
  /nix/store/9nz83nnmjg4m2s5yzqvqc2iikd2n19bf-net-tools-2.10-binlore.drv
  /nix/store/ihjfrqj4vi3z3b994d7b3hdb2pw3aawm-binutils-wrapper-2.40-binlore.drv
  /nix/store/p9i7wgv0f02gp26by2fa563q0h0gz4va-unix-privesc-check-unresholved-1.4-binlore.drv
  /nix/store/sbvi7i3ayrccd85xpsjgr9ljl5vadp3s-ps-procps-3.3.17-binlore.drv
  /nix/store/f3ci3j8frpkmcwzpq2b5hxsh5n2g4h1j-more-binlore.drv
  /nix/store/r8mi91l93rh1srrwfrh2n2hhwc9hjmw1-unix-privesc-check-1.4.drv
these 3 paths will be fetched (6.43 MiB download, 29.85 MiB unpacked):
  /nix/store/xgjcnbg2qj6kly471hgg6bn80bsn43d7-postgresql-15.5
  /nix/store/7f1kq4gn5jy177xjaqy93sbpvyrq7xpl-python2.7-oildev-unstable-2021-07-14
  /nix/store/bpv7938fp2fj4cs7z89ldzgg1248i6va-resholve-0.9.1
copying path '/nix/store/xgjcnbg2qj6kly471hgg6bn80bsn43d7-postgresql-15.5' from 'https://cache.nixos.org'...
copying path '/nix/store/7f1kq4gn5jy177xjaqy93sbpvyrq7xpl-python2.7-oildev-unstable-2021-07-14' from 'https://cache.nixos.org'...
building '/nix/store/0ci8kn552dgki9bvvw4g6j72i1fbxpaq-unix-privesc-check-unresholved-1.4.drv'...
building '/nix/store/ihjfrqj4vi3z3b994d7b3hdb2pw3aawm-binutils-wrapper-2.40-binlore.drv'...
building '/nix/store/9nz83nnmjg4m2s5yzqvqc2iikd2n19bf-net-tools-2.10-binlore.drv'...
building '/nix/store/sbvi7i3ayrccd85xpsjgr9ljl5vadp3s-ps-procps-3.3.17-binlore.drv'...
Running phase: unpackPhase
unpacking source archive /nix/store/1nssihpc7670iz9c5ngw50446w4syvra-unix-privesc-check-1.4.tar.gz
source root is unix-privesc-check-1.4
setting SOURCE_DATE_EPOCH to timestamp 1227470859 of file unix-privesc-check-1.4/unix-privesc-check
Running phase: patchPhase
Running phase: updateAutotoolsGnuConfigScriptsPhase
generating binlore for /nix/store/6zhs433c4cyaih7l65c11zm743sava5a-binutils-wrapper-2.40 by running:
/nix/store/5j4x20ll0n0sw11q7l71aczm15v953jz-yara-4.4.0/bin/yara --scan-list --recursive /nix/store/pj1yyp67mw43x0wib284ywgifll6rink-source/execers.yar <(printf '%s\n' /nix/store/6zhs433c4cyaih7l65c11zm743sava5a-binutils-wrapper-2.40/{bin,lib,libexec}) | /nix/store/angcn8nwmm0yagv2mxpxbc680n4ya2ll-yallback-0.2.0/bin/yallback /nix/store/pj1yyp67mw43x0wib284ywgifll6rink-source/execers.yall
Running phase: configurePhase
generating binlore for /nix/store/i38kh84vidpjjamv69zvvb474m59igv8-net-tools-2.10 by running:
/nix/store/5j4x20ll0n0sw11q7l71aczm15v953jz-yara-4.4.0/bin/yara --scan-list --recursive /nix/store/pj1yyp67mw43x0wib284ywgifll6rink-source/execers.yar <(printf '%s\n' /nix/store/i38kh84vidpjjamv69zvvb474m59igv8-net-tools-2.10/{bin,lib,libexec}) | /nix/store/angcn8nwmm0yagv2mxpxbc680n4ya2ll-yallback-0.2.0/bin/yallback /nix/store/pj1yyp67mw43x0wib284ywgifll6rink-source/execers.yall
error scanning /nix/store/6zhs433c4cyaih7l65c11zm743sava5a-binutils-wrapper-2.40/lib: could not open file
error scanning /nix/store/6zhs433c4cyaih7l65c11zm743sava5a-binutils-wrapper-2.40/libexec: could not open file
no configure script, doing nothing
Running phase: buildPhase
error scanning /nix/store/i38kh84vidpjjamv69zvvb474m59igv8-net-tools-2.10/lib: could not open file
error scanning /nix/store/i38kh84vidpjjamv69zvvb474m59igv8-net-tools-2.10/libexec: could not open file
no Makefile or custom buildPhase, doing nothing
Running phase: installPhase
generating binlore for /nix/store/57c0hxqan3a2m47k4qpwy17fn5208si1-ps-procps-3.3.17 by running:
/nix/store/5j4x20ll0n0sw11q7l71aczm15v953jz-yara-4.4.0/bin/yara --scan-list --recursive /nix/store/pj1yyp67mw43x0wib284ywgifll6rink-source/execers.yar <(printf '%s\n' /nix/store/57c0hxqan3a2m47k4qpwy17fn5208si1-ps-procps-3.3.17/{bin,lib,libexec}) | /nix/store/angcn8nwmm0yagv2mxpxbc680n4ya2ll-yallback-0.2.0/bin/yallback /nix/store/pj1yyp67mw43x0wib284ywgifll6rink-source/execers.yall
using /nix/store/pj1yyp67mw43x0wib284ywgifll6rink-source/overrides/ps- to generate overriden binlore for /nix/store/57c0hxqan3a2m47k4qpwy17fn5208si1-ps-procps-3.3.17
Running phase: fixupPhase
error scanning /nix/store/57c0hxqan3a2m47k4qpwy17fn5208si1-ps-procps-3.3.17/lib: could not open file
error scanning /nix/store/57c0hxqan3a2m47k4qpwy17fn5208si1-ps-procps-3.3.17/libexec: could not open file
shrinking RPATHs of ELF executables and libraries in /nix/store/mpr0h44hzbnqw3h8incdlr1sxc26pwl9-unix-privesc-check-unresholved-1.4
checking for references to /build/ in /nix/store/mpr0h44hzbnqw3h8incdlr1sxc26pwl9-unix-privesc-check-unresholved-1.4...
binlore for /nix/store/i38kh84vidpjjamv69zvvb474m59igv8-net-tools-2.10 written to /nix/store/w7x5lxpbjf1x5fygr7ysvs9y6f24rh5s-net-tools-2.10-binlore
patching script interpreter paths in /nix/store/mpr0h44hzbnqw3h8incdlr1sxc26pwl9-unix-privesc-check-unresholved-1.4
/nix/store/mpr0h44hzbnqw3h8incdlr1sxc26pwl9-unix-privesc-check-unresholved-1.4/bin/unix-privesc-check: interpreter directive changed from "#!/bin/sh" to "/nix/store/cjbyb45nxiqidj95c4k1mh65azn1x896-bash-5.2-p21/bin/sh"
binlore for /nix/store/57c0hxqan3a2m47k4qpwy17fn5208si1-ps-procps-3.3.17 written to /nix/store/8pp74cckg15qh3hp2avkhj88218q3frq-ps-procps-3.3.17-binlore
stripping (with command strip and flags -S -p) in  /nix/store/mpr0h44hzbnqw3h8incdlr1sxc26pwl9-unix-privesc-check-unresholved-1.4/bin
building '/nix/store/p9i7wgv0f02gp26by2fa563q0h0gz4va-unix-privesc-check-unresholved-1.4-binlore.drv'...
binlore for /nix/store/6zhs433c4cyaih7l65c11zm743sava5a-binutils-wrapper-2.40 written to /nix/store/j26fysv150hdgrmr8vfsy0hfprbf3nk5-binutils-wrapper-2.40-binlore
generating binlore for /nix/store/mpr0h44hzbnqw3h8incdlr1sxc26pwl9-unix-privesc-check-unresholved-1.4 by running:
/nix/store/5j4x20ll0n0sw11q7l71aczm15v953jz-yara-4.4.0/bin/yara --scan-list --recursive /nix/store/pj1yyp67mw43x0wib284ywgifll6rink-source/execers.yar <(printf '%s\n' /nix/store/mpr0h44hzbnqw3h8incdlr1sxc26pwl9-unix-privesc-check-unresholved-1.4/{bin,lib,libexec}) | /nix/store/angcn8nwmm0yagv2mxpxbc680n4ya2ll-yallback-0.2.0/bin/yallback /nix/store/pj1yyp67mw43x0wib284ywgifll6rink-source/execers.yall
error scanning /nix/store/mpr0h44hzbnqw3h8incdlr1sxc26pwl9-unix-privesc-check-unresholved-1.4/lib: could not open file
error scanning /nix/store/mpr0h44hzbnqw3h8incdlr1sxc26pwl9-unix-privesc-check-unresholved-1.4/libexec: could not open file
binlore for /nix/store/mpr0h44hzbnqw3h8incdlr1sxc26pwl9-unix-privesc-check-unresholved-1.4 written to /nix/store/ghy7wgh1nwrjr0n6ns27b6d87na63d7n-unix-privesc-check-unresholved-1.4-binlore
copying path '/nix/store/bpv7938fp2fj4cs7z89ldzgg1248i6va-resholve-0.9.1' from 'https://cache.nixos.org'...
building '/nix/store/3s8q8yr21pz1wqpw0iczfv1cdyvnjzpq-postgresql-15.5-binlore.drv'...
generating binlore for /nix/store/xgjcnbg2qj6kly471hgg6bn80bsn43d7-postgresql-15.5 by running:
/nix/store/5j4x20ll0n0sw11q7l71aczm15v953jz-yara-4.4.0/bin/yara --scan-list --recursive /nix/store/pj1yyp67mw43x0wib284ywgifll6rink-source/execers.yar <(printf '%s\n' /nix/store/xgjcnbg2qj6kly471hgg6bn80bsn43d7-postgresql-15.5/{bin,lib,libexec}) | /nix/store/angcn8nwmm0yagv2mxpxbc680n4ya2ll-yallback-0.2.0/bin/yallback /nix/store/pj1yyp67mw43x0wib284ywgifll6rink-source/execers.yall
error scanning /nix/store/xgjcnbg2qj6kly471hgg6bn80bsn43d7-postgresql-15.5/libexec: could not open file
binlore for /nix/store/xgjcnbg2qj6kly471hgg6bn80bsn43d7-postgresql-15.5 written to /nix/store/d8sazlp5iqi63jky90nwi569jnfrfkw5-postgresql-15.5-binlore
building '/nix/store/f3ci3j8frpkmcwzpq2b5hxsh5n2g4h1j-more-binlore.drv'...
substituteStream(): WARNING: '--replace' is deprecated, use --replace-{fail,warn,quiet}. (file '/nix/store/cisqp7pslm5032cp9gy18nfdlp5an34g-more-binlore/execers')
substituteStream(): WARNING: pattern /nix/store/mpr0h44hzbnqw3h8incdlr1sxc26pwl9-unix-privesc-check-unresholved-1.4/ doesn't match anything in file '/nix/store/cisqp7pslm5032cp9gy18nfdlp5an34g-more-binlore/wrappers'
building '/nix/store/r8mi91l93rh1srrwfrh2n2hhwc9hjmw1-unix-privesc-check-1.4.drv'...
Running phase: unpackPhase
unpacking source archive /nix/store/mpr0h44hzbnqw3h8incdlr1sxc26pwl9-unix-privesc-check-unresholved-1.4
source root is unix-privesc-check-unresholved-1.4
Running phase: patchPhase
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: installPhase
Running phase: fixupPhase
[resholve context] : invoking resholve with PWD=/nix/store/95vgpg6hq829fhqbykiifzav6pf3bzfg-unix-privesc-check-1.4
[resholve context] RESHOLVE_LORE=/nix/store/cisqp7pslm5032cp9gy18nfdlp5an34g-more-binlore
[resholve context] RESHOLVE_EXECER='cannot:/nix/store/dvvb6frpdnimidx1f51zjgi3af8rlny1-glibc-2.38-27-bin/bin/ldd cannot:/nix/store/xgjcnbg2qj6kly471hgg6bn80bsn43d7-postgresql-15.5/bin/psql cannot:/nix/store/4idjhhrwky07qj8bq7xml7gi9frw4ss5-openssh-9.6p1/bin/ssh-add cannot:/nix/store/c9rqpsd8b3l82z0fjz90n7x28nkc0240-util-linux-2.39.3-bin/bin/swapon'
[resholve context] RESHOLVE_FAKE='external:'\''lanscan'\'';'\''mount'\'';'\''passwd'\'''
[resholve context] RESHOLVE_INPUTS=/nix/store/w48cndp5bwz4x4l49yr2gbz09g6f91dq-gawk-5.2.2/bin:/nix/store/cjbyb45nxiqidj95c4k1mh65azn1x896-bash-5.2-p21/bin:/nix/store/6zhs433c4cyaih7l65c11zm743sava5a-binutils-wrapper-2.40/bin:/nix/store/5idwbbv23b6vnqdicx97s3hsgrwwnj7j-coreutils-9.4/bin:/nix/store/kv5fs6gk153xkg4qzrnk0cs8myrc1wki-file-5.45/bin:/nix/store/4ajik70nplhkb8ndn3gqh7v0b09hmvg9-findutils-4.9.0/bin:/nix/store/dvvb6frpdnimidx1f51zjgi3af8rlny1-glibc-2.38-27-bin/bin:/nix/store/6i00hdmzlj56qy500p5gb5v88wfj6nhg-gnugrep-3.11/bin:/nix/store/9zial3lqry9f7rsw31r7vs5p1mnb7lan-gnused-4.9/bin:/nix/store/i38kh84vidpjjamv69zvvb474m59igv8-net-tools-2.10/bin:/nix/store/4idjhhrwky07qj8bq7xml7gi9frw4ss5-openssh-9.6p1/bin:/nix/store/xgjcnbg2qj6kly471hgg6bn80bsn43d7-postgresql-15.5/bin:/nix/store/57c0hxqan3a2m47k4qpwy17fn5208si1-ps-procps-3.3.17/bin:/nix/store/c9rqpsd8b3l82z0fjz90n7x28nkc0240-util-linux-2.39.3-bin/bin:/nix/store/czis9iqs3ph8rxkndks4lf2qgh752yf4-which-2.21/bin
[resholve context] RESHOLVE_INTERPRETER=/nix/store/cjbyb45nxiqidj95c4k1mh65azn1x896-bash-5.2-p21/bin/bash
[resholve context] /nix/store/bpv7938fp2fj4cs7z89ldzgg1248i6va-resholve-0.9.1/bin/resholve --overwrite bin/unix-privesc-check
Overwrote '/nix/store/95vgpg6hq829fhqbykiifzav6pf3bzfg-unix-privesc-check-1.4/bin/unix-privesc-check'
shrinking RPATHs of ELF executables and libraries in /nix/store/95vgpg6hq829fhqbykiifzav6pf3bzfg-unix-privesc-check-1.4
checking for references to /build/ in /nix/store/95vgpg6hq829fhqbykiifzav6pf3bzfg-unix-privesc-check-1.4...
patching script interpreter paths in /nix/store/95vgpg6hq829fhqbykiifzav6pf3bzfg-unix-privesc-check-1.4
stripping (with command strip and flags -S -p) in  /nix/store/95vgpg6hq829fhqbykiifzav6pf3bzfg-unix-privesc-check-1.4/bin
/nix/store/95vgpg6hq829fhqbykiifzav6pf3bzfg-unix-privesc-check-1.4
```
```
$ ./result/bin/unix-privesc-check 
unix-privesc-check v1.4 ( http://pentestmonkey.net/tools/unix-privesc-check )

Usage: unix-privesc-check { standard | detailed }

"standard" mode: Speed-optimised check of lots of security settings.

"detailed" mode: Same as standard mode, but also checks perms of open file
                 handles and called files (e.g. parsed from shell scripts,
                 linked .so files).  This mode is slow and prone to false 
                 positives but might help you find more subtle flaws in 3rd
                 party programs.

This script checks file permissions and other settings that could allow
local users to escalate privileges.

Use of this script is only permitted on systems which you have been granted
legal permission to perform a security assessment of.  Apart from this 
condition the GPL v2 applies.

Search the output for the word 'WARNING'.  If you don't see it then this
script didn't find any problems.
```
```
$ ./result/bin/unix-privesc-check standard
Assuming the OS is: linux
/nix/store/czis9iqs3ph8rxkndks4lf2qgh752yf4-which-2.21/bin/which: no file in (/run/wrappers/bin:/home/sebastian/.nix-profile/bin:/nix/profile/bin:/home/sebastian/.local/state/nix/profile/bin:/etc/profiles/per-user/sebastian/bin:/nix/var/nix/profiles/default/bin:/run/current-system/sw/bin:/usr/bin:/bin:/sbin:/usr/sbin)
ERROR: Dependend program 'file' is mising.  Can't run.  Sorry!
```

**Comment by D3vil0p3r** [comment]
It is weird. I'm launching the same command but I get no errors:
```
./result/bin/unix-privesc-check standard

Assuming the OS is: linux
Starting unix-privesc-check v1.4 ( http://pentestmonkey.net/tools/unix-privesc-check )

This script checks file permissions and other settings that could allow
local users to escalate privileges.

Use of this script is only permitted on systems which you have been granted
legal permission to perform a security assessment of.  Apart from this 
condition the GPL v2 applies.

Search the output below for the word 'WARNING'.  If you don't see it then
this script didn't find any problems.


############################################
Recording hostname
############################################
nixos

############################################
Recording uname
############################################
Linux nixos 6.1.77 #1-NixOS SMP PREEMPT_DYNAMIC Mon Feb  5 20:13:03 UTC 2024 x86_64 GNU/Linux
...
```
@K900 do you have an idea why @MinerSebas is getting a different behavior?

**Comment by MinerSebas** [comment]
Alternatively could you run `which` in your shell, to check whether your `Path` already contains the `file` and `strings` programs.
```
$ which which
which: shell built-in command
$ which file
file not found
$ which strings
strings not found
```

**Comment by MinerSebas** [comment]
I think i found the problem.
Its the following lines:
```
# Check dependent programs are installed
# Assume "which" is installed!
PROGS="ls awk grep cat mount xargs file ldd strings"
for PROG in $PROGS; do
	/nix/store/czis9iqs3ph8rxkndks4lf2qgh752yf4-which-2.21/bin/which $PROG 2>&1 > /dev/null
	if [ ! $? = "0" ]; then
		echo "ERROR: Dependend program '$PROG' is mising.  Can't run.  Sorry!"
		exit 1
	fi
done
```

resholve hardcodes all called Programa to their location in the store, without modifying `PATH`.
The Problem in the above snippet is, that the script is calling `which`, but not the programs in `PROGS`.
The reason @D3vil0p3r had no Problems here, is that all the programs were already in his `PATH`.

When I remove the above snippet, the script works correctly for me, as when actually running the binaries from `PROGS`, they were already hardcoded by resholve.

Conclusion: Patch the above code snippet out, as it isn't necessary here.

**Review by MinerSebas — APPROVED** [review]
Everything works for me now.

**Pull request merged**